src/debug/ee/arm64/arm64walker.cpp

```diff
@@ -2,21 +2,460 @@
+        }
+        return TRUE;
+    }
+
+    //Conditional Branches
+    _ASSERTE(context != NULL);
+   
+
+    if ((opcode & 0xFF000010) == 0x54000000) // B.cond
+    {
+        WORD cond = opcode & 0xF;
+        bool result = false;
+        switch (cond >> 1)
+        {
+        case 0x0:  result = (context->Cpsr & NZCV_Z) != 0; // EQ or NE
+            break;
+        case 0x1:  result = (context->Cpsr & NZCV_C) != 0; // CS or CC
+            break;
+        case 0x2:  result = (context->Cpsr & NZCV_N) != 0; // MI or PL
+            break;
+        case 0x3:  result = (context->Cpsr & NZCV_V) != 0; // VS or VC
+            break;
+        case 0x4:  result = ((context->Cpsr & NZCV_C) != 0) && ((context->Cpsr & NZCV_Z) == 0); // HI or LS
+            break;
+        case 0x5:  result = ((context->Cpsr & NZCV_N) >> NZCV_N_BIT) == ((context->Cpsr & NZCV_V) >> NZCV_V_BIT); // GE or LT
+            break;
+        case 0x6:  result = ((context->Cpsr & NZCV_N) >> NZCV_N_BIT) == ((context->Cpsr & NZCV_V) >> NZCV_V_BIT) && ((context->Cpsr & NZCV_Z) == 0); // GT or LE
+            break;
+        case 0x7:  result = true; // AL
+            break;
+        }
+
+        if ((cond & 0x1) && (cond & 0xF) != 0) { result = !result; }
+
+        if (result)
+        {
+            walk = WALK_BRANCH;
+            offset = ((opcode >> 5) & 0x7FFFF)<<2; //imm19:00 -> 21 bits
+
+            if (offset & 0x100000) //sign extension check for 21's bit
+            {
+                offset = offset | 0xFFFFFFFFFFE00000;
+            }
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to B.cond %p \n", opcode, offset));
+        }
+        else // NOP
+        {
+            walk = WALK_UNKNOWN;
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to B.cond but evaluated as NOP \n", opcode));
+            offset = MAX_INSTRUCTION_LENGTH; 
+        }
+        
+        return TRUE;
+
+    }
+
+    
+    int RegNum       = opcode & 0x1F;
+    PCODE RegContent = GetReg(context, RegNum);
+
+    if ((opcode & 0xFE000000) == 0xB4000000) // CBNZ || CBZ
+    {
+        bool result = false;
+        if (opcode & 0x01000000) //CBNZ
+        {
+            result = RegContent != 0;
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to CBNZ X%d \n", opcode, RegNum));
+        }
+        else //CBZ
+        {
+            result = RegContent == 0;
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to CBZ X%d \n", opcode, RegNum));
+        }
+        
+        if (result)
+        {
+            walk = WALK_BRANCH;
+            offset = ((opcode >> 5) & 0x7FFFF) << 2; //imm19:00 -> 21 bits
+            if (offset & 0x100000) //sign extension check for 21'th bit
+            {
+                offset = offset | 0xFFFFFFFFFFE00000;
+            }
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to CB[N]Z X%d %p \n", opcode, RegNum, offset));
```

Comment:
The above would be useful if it were extracted to a function as LDR literal needs to do this as well